src/mesh/generic/Boundary.cpp

```diff
@@ -79,4 +93,64 @@ double Boundary::interpolate_prescribed_load(std::vector<std::pair<double, doubl
 
     return (1.0 - interpolation_factor) * load_0 + interpolation_factor * load_1;
 }
+
+void Boundary::generate_sinusoidal(json const& boundary,
+                                   std::string const& name,
+                                   double generate_time_step)
+{
+    using namespace ranges;
+
+    for (auto const& mandatory_field : {"Period", "Phase", "NumberOfCycles"})
+    {
+        if (!boundary.count(mandatory_field))
+        {
+            throw std::runtime_error("\"" + std::string(mandatory_field)
+                                     + "\" was not specified in \"BoundaryCondition\".");
+        }
+    }
+
+    std::vector<double> amplitude = boundary[name];
+    std::vector<double> period = boundary["Period"];
+    std::vector<double> number_of_cycles = boundary["NumberOfCycles"];
+    std::vector<double> phase = boundary["Phase"];
+
```

Comment:
Can these be marked `const`?  You might also be able to use a `{}` initialisation `... amplitiude{boundary[name]};`